recipes/Tedlium2/tedlium2_prepare.py

```diff
@@ -0,0 +1,200 @@
+
+        # clip and save the current utterance
+        clipped_save_path = f"{utt_save_folder}/{talk_id}-{str(i)}.wav"
+        # we avoid duplicated clip and save
+        if not os.path.exists(clipped_save_path):
+            start = float(line[3]) * sample_rate
+            end = float(line[4]) * sample_rate
+            curr_utt = original_speech[:, int(start) : int(end)]
+            torchaudio.save(clipped_save_path, curr_utt, sample_rate)
+        # append to the csv entry list
+        csv_line = [
+            f"{talk_id}-{str(i)}",
+            str(duration),
+            clipped_save_path,
+            spk_id,
+            transcript,
+        ]
+        entry.append(csv_line)
+
+    # create the csv file for the utterences in the talk recording.
+    # the csv files for all the talks will be merged
+    csv_file = f"{csv_save_folder}/{talk_id}.csv"
+    csv_output = [["ID", "duration", "wav", "spk_id", "wrd"]]
+    csv_output = csv_output + entry
+    with open(csv_file, mode="w") as csv_f:
+        csv_writer = csv.writer(
+            csv_f, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+        )
+        for line in csv_output:
+            csv_writer.writerow(line)
+
+
+def prepare_tedlium2(
+    data_folder, utt_save_folder, csv_save_folder, skip_prep=False
+):
+    """
+    This function prepares the Tedlium2 dataset.
+    Download link: https://lium.univ-lemans.fr/ted-lium2/
+
+    Arguments
+    ---------
+    data_folder : str
+        Path to the folder where the original Tedlium2 dataset is stored.
+    utt_save_folder : list
+        Path where to save the clipped utterence-leve recordings.
+    csv_save_folder: str
+        Path where to save the generated .csv files.
+    skip_prep: bool
+        If True, data preparation is skipped.
+
+    Example
+    -------
+    >>> data_folder = 'datasets/TEDLIUM_release2'
+    >>> utt_save_folder = 'datasets/TEDLIUM_release2_processed'
+    >>> csv_save_folder = 'TEDLIUM2'
+    >>> prepare_tedlium2(data_folder, utt_save_folder, csv_save_folder)
+
+    """
+    if skip_prep:
+        return
+
+    splits = [
+        "train",
+        "dev",
+        "test",
+    ]
+    for split in splits:
+        utt_save_folder_split = f"{utt_save_folder}/{split}"
+        csv_save_folder_split = f"{csv_save_folder}/{split}"
+        os.makedirs(utt_save_folder_split, exist_ok=True)
+        os.makedirs(csv_save_folder_split, exist_ok=True)
+        new_filename = os.path.join(csv_save_folder_split, split) + ".csv"
+        if os.path.exists(new_filename):
+            continue
+        logger.info("Preparing %s..." % new_filename)
+        talk_sphs = os.listdir(f"{data_folder}/{split}/sph")
+        processes = []
+
+        for talk_sph in talk_sphs:
+            talk_name = talk_sph[:-4]
+            talk_sph_path = f"{data_folder}/{split}/sph/{talk_sph}"
+            talk_stm_path = f"{data_folder}/{split}/stm/{talk_name}.stm"
+            p = multiprocessing.Process(
```

Comment:
why are you not using `parell_map` ? See librispeech_prepary.py and `speechbrain.utils.parallel`